ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -0,0 +1,226 @@
+ * container needs to be specified when posting a blob. Container id is part of the blob’s properties, and cannot
+ * be modified once the blob is created.
+ *
+ * Container name is provided by a user as an external reference to that container under an {@link Account}. Container
+ * id is one-to-one mapped to container name, and serves as an internal identifier of a container. Container name/id
+ * has to be distinct within the same {@link Account}, but can be the same across different {@link Account}s. Container
+ * metadata is made in JSON, which is generic to contain additional information of the metadata.
+ *
+ * Version 1 of Container metadata in JSON is in the format below.
+ *  {
+ *    "containerName": "MyPrivateContainer",
+ *    "description": "This is my private container",
+ *    "isPrivate": "true",
+ *    "containerId": 0,
+ *    "version": 1,
+ *    "status": "active"
+ *  }
+ */
+public class Container {
+  // static variables
+  public static final String CONTAINER_METADATA_VERSION_KEY = "version";
+  public static final String CONTAINER_NAME_KEY = "containerName";
+  public static final String CONTAINER_ID_KEY = "containerId";
+  public static final String CONTAINER_STATUS_KEY = "status";
+  public static final String CONTAINER_DESCRIPTION_KEY = "description";
+  public static final String CONTAINER_IS_PRIVATE_KEY = "isPrivate";
+  public static final String CONTAINER_STATUS_ACTIVE = "active";
+  public static final String CONTAINER_STATUS_INACTIVE = "inactive";
+  public static final short CONTAINER_METADATA_VERSION_1 = 1;
+  // container field variables
+  private final short id;
+  private final String name;
+  private final String status;
+  private final String description;
+  private final boolean isPrivate;
+  private final Account parentAccount;
+  private final JSONObject metadata;
+
+  /**
+   * Constructor from container metadata.
+   * @param metadata The metadata of the container in JSON.
+   * @param parentAccount The parent {@link Account} of the container.
+   */
+  public Container(JSONObject metadata, Account parentAccount) throws JSONException {
+    if (metadata == null) {
+      throw new IllegalArgumentException("metadata cannot be null.");
+    }
+    if (parentAccount == null) {
+      throw new IllegalArgumentException("parentAccount cannot be null.");
+    }
+    short containerMetadataVersion = (short) metadata.getInt(CONTAINER_METADATA_VERSION_KEY);
+    switch (containerMetadataVersion) {
+      case CONTAINER_METADATA_VERSION_1:
+        this.id = (short) metadata.getInt(CONTAINER_ID_KEY);
+        this.name = metadata.getString(CONTAINER_NAME_KEY);
+        this.status = metadata.getString(CONTAINER_STATUS_KEY);
+        this.description = metadata.optString(CONTAINER_DESCRIPTION_KEY);
+        this.isPrivate = metadata.getBoolean(CONTAINER_IS_PRIVATE_KEY);
+        this.metadata = metadata;
+        this.parentAccount = parentAccount;
```

Comment:
what is the control flow wrt adding a new container to an Acccount. I was expecting parentAccount.addContainerAndMetadata(this) here. Or is the caller expected to call 
```
Account account1= new Account(.....)
Container container1 = new Container(..., account1)
account1.addContainerAndMetadata(container1);
```